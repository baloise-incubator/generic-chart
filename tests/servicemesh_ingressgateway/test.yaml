templates:
  - deployment.yaml
#  - external-secret.yaml
#  - ingress-route.yaml
  - istio-igw-withvs.yaml
#  - pvc.yaml
  - service.yaml
  - serviceaccount.yaml
#  - serviceloadbalancer.yaml
#  - servicemonitor.yaml
#  - statefulset-ingress-route.yaml
#  - statefulset-service.yaml
#  - virtualservice.yaml
tests:
  - release:
      name: simple-app
      namespace: namespace-test
    values:
      - values.yaml
    asserts:
      - matchSnapshot: {}
