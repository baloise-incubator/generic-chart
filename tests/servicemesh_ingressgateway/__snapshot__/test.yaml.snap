"":
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: generic-chart
        helm.sh/chart: generic-chart-3.9.3
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      replicas: 1
      revisionHistoryLimit: 1
      selector:
        matchLabels:
          app: generic-chart
          release: simple-app
      strategy:
        type: RollingUpdate
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            app: generic-chart
            release: simple-app
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
              - labelSelector:
                  matchExpressions:
                  - key: app
                    operator: In
                    values:
                    - generic-chart
                  - key: release
                    operator: In
                    values:
                    - simple-app
                topologyKey: kubernetes.io/hostname
          automountServiceAccountToken: true
          containers:
          - image: registry.example.tld/org/repo:v1
            imagePullPolicy: IfNotPresent
            name: generic-chart
            ports:
            - containerPort: 8080
              name: http
              protocol: TCP
            resources: {}
            securityContext: {}
          securityContext: {}
          serviceAccountName: generic-chart
          topologySpreadConstraints:
          - labelSelector:
              matchLabels:
                app: generic-chart
                release: simple-app
            maxSkew: 1
            topologyKey: topology.kubernetes.io/any
            whenUnsatisfiable: DoNotSchedule
  2: |
    apiVersion: networking.istio.io/v1beta1
    kind: VirtualService
    metadata:
      labels:
        app: generic-chart
        helm.sh/chart: generic-chart-3.9.3
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      gateways:
      - generic-chart
      hosts:
      - example.example.tld
      http:
      - route:
        - destination:
            host: generic-chart
            port:
              number: 8080
  3: |
    apiVersion: networking.istio.io/v1beta1
    kind: Gateway
    metadata:
      labels:
        app: generic-chart
        helm.sh/chart: generic-chart-3.9.3
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      selector:
        istio: ingressgateway
      servers:
      - hosts:
        - example.example.tld
        port:
          name: generic-chart
          number: 80
          protocol: HTTP
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: generic-chart
        helm.sh/chart: generic-chart-3.9.3
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      ports:
      - name: http
        port: 8080
        protocol: TCP
        targetPort: http
      selector:
        app: generic-chart
        release: simple-app
      type: ClusterIP
  5: |
    apiVersion: v1
    automountServiceAccountToken: false
    kind: ServiceAccount
    metadata:
      labels:
        app: generic-chart
        helm.sh/chart: generic-chart-3.9.3
        release: simple-app
      name: generic-chart
      namespace: namespace-test
